homeassistant/components/rest/data.py

```diff
@@ -107,6 +107,7 @@ async def async_update(self, log_errors: bool = True) -> None:
                 timeout=self._timeout,
                 follow_redirects=True,
             )
+            response.encoding = self._encoding
```

Comment:
Looks strange. Shouldnt it be optional?

Reply:
I agree it looks strange. Please also add a test which fails without this change and passes with it

Reply:
Ok, so I have no idea how to showcase this.
The issue was from [this webpage](https://intranet.chmi.cz/files/portal/docs/meteo/om/predpovedi/CR/p_CR.html) that even if content type is set to `windows-1250` it comes back as `utf-8` so therefore even if stating a default encoding of `windows-1250` [when setting up the connection](https://github.com/home-assistant/core/pull/119003/files#diff-be7a4c3e1960d55635a16a0bd35f405baab4befc77e820bbdd9cf17872ef5365R91) it will go to `utf-8` as the content type header is returned as such.

What the change does is to change the encoding of the return before reading it's content.
I would assume that if the user provides an encoding we would return it as such?

Reply:
The point being. It should be possible to not force an encoding. It should respect what the server sends.

HA should convert the encoding the server provided to internal strings (ie utf8).

Forcing encoding should mainly modify the request, setting what is and accepted request encoding.

Ps.. I might be missing something

Reply:
I checked that server. It's obviously wrongly configured. It's even if you override the charset to windows-1250 it looks invalid.

So what this pull is trying to do is override what the server responds with. I can understand that. You do need to set up a testcase for it. Also we can't do this by modifying httpx variables on the response object. This must be done by grabbing the raw data from httpx, and decoding it using the forced charset.

Reply:
> Also we can't do this by modifying httpx variables on the response object

Why? Can't seem to find again where I read it but it works fine and I don't see the impact of anything else?

The question for me is maybe more if we need another config parameter for `force_encoding` as rightfully pointed out is not going to be needed that often.

Can obviously try to setup some testcase once we agree how it's supposed to work.